software/base/src/main/java/brooklyn/entity/brooklynnode/BrooklynNodeImpl.java

```diff
@@ -192,7 +198,8 @@ protected void postStop() {
         super.postStop();
         ConfigBag stopParameters = BrooklynTaskTags.getCurrentEffectorParameters();
         //unmanage only if stopping the machine
-        if (stopParameters == null || stopParameters.get(StopSoftwareParameters.STOP_MACHINE)) {
+        if (stopParameters == null || (StopMode.NEVER != stopParameters.get(StopSoftwareParameters.STOP_MACHINE_MODE) 
+                && Boolean.TRUE.equals(stopParameters.get(StopSoftwareParameters.STOP_MACHINE)))) {
```

Comment:
The combination of `STOP_MACHINE` and `STOP_MACHINE_MODE` is confusing! If this specific permutation gets checked in multiple places, it might be worth providing a convenience method.